packages/lib/src/vis/scatter/hooks.ts

```diff
@@ -1,66 +1,6 @@
-import type { Domain, NumArray, ScaleType } from '@h5web/shared';
-import { rgb } from 'd3-color';
-import { useMemo } from 'react';
+import { createMemo } from '@h5web/shared';
 
-import type { ColorMap } from '../heatmap/models';
-import { getInterpolator } from '../heatmap/utils';
-import { useVisCanvasContext } from '../shared/VisCanvasProvider';
-import { createAxisScale } from '../utils';
+import { getValueToColor, getIndexToPosition } from './utils';
 
-const CAMERA_FAR = 1000; // R3F's default
-
-export function useBufferAttributes(
-  abscissas: NumArray,
-  ordinates: NumArray,
-  data: NumArray,
-  dataToColorScale: (val: number) => [number, number, number]
-) {
-  const { abscissaScale, ordinateScale } = useVisCanvasContext();
-
-  return useMemo(() => {
-    const position = new Float32Array(3 * data.length);
-    const color = new Uint8Array(3 * data.length);
-
-    data.forEach((val, index) => {
-      color.set(dataToColorScale(val), 3 * index);
-
-      const x = abscissaScale(abscissas[index]);
-      const y = ordinateScale(ordinates[index]);
-
-      const hasFiniteCoords = Number.isFinite(x) && Number.isFinite(y);
-
-      /* Render points with NaN/Infinity coordinates (i.e. values <= 0 in log)
-       * at origin to avoid Three warning, and outside of camera's field of view
-       * to hide them and any segments connecting them. */
-      position.set([x, y, hasFiniteCoords ? 0 : CAMERA_FAR], 3 * index);
```

Comment:
There was a bug here: we weren't doing the "at origin" part like in `DataCurve`.

Reply:
Ah, because the comment seems to indicate that it's the moving of the point to the origin that removes the warning, (and that moving it behind the camera is just for hiding the resulting line segments). I think this is right, since the coordinates in the `position` attribute are passed directly to the vertex shader, and it seems logical to me that the shader wouldn't like dealing with infinite coordinates.

Reply:
Ha! Yes, you are right ! :+1: